src/events.js

```diff
@@ -1076,28 +1076,38 @@ const redirectSpecimenPage = async (accessionID1, accessionID3, selectedVisit, f
     specimenTemplate(data, formData);
 }
 
-export const addEventBiospecimenCollectionForm = async (participantData, biospecimenData) => {
+export const addEventBiospecimenCollectionFormSave = async (participantData, biospecimenData) => {
     const collectionSaveExit = document.getElementById('collectionSave');
-    collectionSaveExit.addEventListener('click', async () => {
-        try {
-            await collectionSubmission(participantData, biospecimenData);
-        } catch (error) {
-            console.error("Error saving collection: ", error);
-            showNotifications({ title: 'Error saving collection!', body: error.message });
-        }
+    collectionSaveExit && collectionSaveExit.addEventListener('click', async () => {
+        await checkFormAndSave(participantData, biospecimenData, false);
     });
 
     const collectionSaveContinue = document.getElementById('collectionNext');
-    collectionSaveContinue.addEventListener('click', async () => {
-        try {
-            await collectionSubmission(participantData, biospecimenData, true);
-        } catch (error) {
-            console.error("Error saving collection: ", error);
-            showNotifications({ title: 'Error saving collection!', body: error.message });
-        }
+    collectionSaveContinue && collectionSaveContinue.addEventListener('click', async () => {
+        await checkFormAndSave(participantData, biospecimenData, true);
     });
 };
 
+/**
+ * Validate the form and save the collection.
+ * @param {Object} participantData - Participant data object.
+ * @param {Object} biospecimenData - Biospecimen data object.
+ * @param {Boolean} shouldNavigateToReview - Boolean to indicate the button clicked. false -> 'save' (stay on page). true -> 'Go to Review' (navigate to Review screen).
+ */
+const checkFormAndSave = async (participantData, biospecimenData, shouldNavigateToReview) => {
+    try {
+        const inputFields = Array.from(document.getElementsByClassName('input-barcode-id'));
+        const isFormDataValid = inputFields.every(input => validateFormInputField(input, biospecimenData, true));
+        
+        isFormDataValid ?
+            await collectionSubmission(participantData, biospecimenData, shouldNavigateToReview) :
+            showTimedNotifications({ title: 'Data Errors Exist!', body: 'Please correct data entry errors in red before saving.' });
+    } catch (error) {
+        console.error("Error saving collection: ", error);
```

Comment:
Brian had a good question regarding `console.error()` earlier today... Do we know yet if these will be visible in DataDog? Or do we need to add something similar to what was done in the PWA?

Reply:
The console.error logs and the custom logs in the PWA both go to DataDog but neither is visible in the main error reporting screen. That main screen only shows uncaught errors.

The path to view console.error and custom logs is:
UX Monitoring -> Sessions Explorer -> Sort descending by error count.
Then click into the individual session to see console.error and custom logs.